### DISCUSSION TITLE
How to activate normal mode in neo-tree rename pop-up?

### DISCUSSION THREAD

**Discussion by sgoudham** [nvim-neo-tree/neo-tree.nvim#1039] [Q&A] [answered]
Hey :wave: 

I'm struggling to figure out what handles the rename popup inside of neo-tree so I can (hopefully) configure it so that pressing escape activates normal mode instead of quitting the pop-up entirely. I had a look at the full configuration and couldn't see any relevant options for this. 

I'm using [LazyVim](https://www.lazyvim.org/plugins/editor#neo-treenvim)'s defaults. For convenience, here is the full configuration that is being used:

```lua
{
  "nvim-neo-tree/neo-tree.nvim",
  cmd = "Neotree",
  keys = {
    {
      "<leader>fe",
      function()
        require("neo-tree.command").execute({ toggle = true, dir = require("lazyvim.util").get_root() })
      end,
      desc = "Explorer NeoTree (root dir)",
    },
    {
      "<leader>fE",
      function()
        require("neo-tree.command").execute({ toggle = true, dir = vim.loop.cwd() })
      end,
      desc = "Explorer NeoTree (cwd)",
    },
    { "<leader>e", "<leader>fe", desc = "Explorer NeoTree (root dir)", remap = true },
    { "<leader>E", "<leader>fE", desc = "Explorer NeoTree (cwd)", remap = true },
  },
  deactivate = function()
    vim.cmd([[Neotree close]])
  end,
  init = function()
    vim.g.neo_tree_remove_legacy_commands = 1
    if vim.fn.argc() == 1 then
      local stat = vim.loop.fs_stat(vim.fn.argv(0))
      if stat and stat.type == "directory" then
        require("neo-tree")
      end
    end
  end,
  opts = {
    sources = { "filesystem", "buffers", "git_status", "document_symbols" },
    open_files_do_not_replace_types = { "terminal", "Trouble", "qf", "Outline" },
    filesystem = {
      bind_to_cwd = false,
      follow_current_file = true,
      use_libuv_file_watcher = true,
    },
    window = {
      mappings = {
        ["<space>"] = "none",
      },
    },
    default_component_configs = {
      indent = {
        with_expanders = true, -- if nil and file nesting is enabled, will enable expanders
        expander_collapsed = "",
        expander_expanded = "",
        expander_highlight = "NeoTreeExpander",
      },
      icon = {
        folder_empty = "󰜌",
        folder_empty_open = "󰜌",
      },
      git_status = {
        symbols = {
          renamed = "󰁕",
          unstaged = "󰄱",
        },
      },
    },
  },
  config = function(_, opts)
    require("neo-tree").setup(opts)
    vim.api.nvim_create_autocmd("TermClose", {
      pattern = "*lazygit",
      callback = function()
        if package.loaded["neo-tree.sources.git_status"] then
          require("neo-tree.sources.git_status").refresh()
        end
      end,
    })
  end,
}
```

It's worth mentioning that `dressing.nvim` has the option to enable this behaviour by setting `insert_only = false`. I tried this and it only worked for the LSP rename popup and not for neo-tree.

I initially raised this inside of LazyVim but I realised I might have been barking down the wrong tree so I figured I'd raise it here before going back (https://github.com/LazyVim/LazyVim/discussions/1077)

`Screenshot of Popup`
![image](https://github.com/nvim-neo-tree/neo-tree.nvim/assets/58985301/f5a66ae7-243d-48f3-96e0-672f417e1825)

**Comment by LoneExile** [comment]
I'm also interested in this native solution. So far, I've been using tmux to copy text instead.

**Reply by sgoudham** [reply]
So interestingly @LoneExile, I was looking at the new v3.0 milestone and this issue #1028 stood out to me for the option name. It sounds like an option that would solve this. 

I searched it cause I was wondering if I missed it in the README but turns it out [it's only present in the core UI logic and in the defaults.lua file](https://github.com/search?q=repo%3Anvim-neo-tree%2Fneo-tree.nvim+close_floats_on_escape_key&type=code).

I've added this to my config now but it still doesn't seem to work, are you able to get this working?

(Or maybe this is another option entirely and I'm misunderstanding something!)

```lua
  {
    "nvim-neo-tree/neo-tree.nvim",
    opts = {
      close_floats_on_escape_key = false,
    },
  },
```

**Reply by LoneExile** [reply]
[#1043](https://github.com/nvim-neo-tree/neo-tree.nvim/pull/1043) like this right? @sgoudham

**Comment by sgoudham** [comment, marked as answer]
Closing as #1043 has been merged into main: https://github.com/nvim-neo-tree/neo-tree.nvim/commit/a64419ddeafdf6adcab7d2406819df2c477fda23

Thanks again for raising that PR! @LoneExile
